js/app.js

```diff
@@ -13,11 +13,82 @@ Calculator.prototype.getHistoryAsString = function() {
 
 Calculator.prototype.add = function(num1, num2) {
     // 1. zamień wartości przekazane przez parametr na typ number
+    num1 = Number(num1);
+    num2 = Number(num2);
     // 2. sprawdź czy są one poprawne
+    if (isNaN(num1) || isNaN(num2)) {
+        alert('Podane wartości są niepoprawne');
+        return;
+    }
     // 3. jeśli tak to wykonaj działanie i zapisz jego resultat
+    const result = num1 + num2;
     // 4. dodaj do historii operacji to działanie w fomie: 1 + 1 = 2
+    this.history.push(`${num1} + ${num2} = ${result}`);
 }
 
+Calculator.prototype.subtract = function(num1, num2) {
+    num1 = Number(num1);
+    num2 = Number(num2);
+    
+    if (isNaN(num1) || isNaN(num2)) {
+        alert('Podano niepoprawne wartości. Spróbuj ponownie.');
+        return;
+    }
```

Comment:
Zwróć uwagę, że ten kod się powtarza - warto przenieść sprawdzenie w inne miejsce, aby to zrobić przed uruchamianiem metod "obliczających"